src/asammdf/blocks/conversion_utils.py

```diff
@@ -228,287 +224,308 @@ def conversion_transfer(
                         "conversion_type": v4c.CONVERSION_TYPE_RTABX,
-                        "default_addr": conversion.referenced_blocks["default_addr"],
+                        "default_addr": typing.cast(bytes, conversion.referenced_blocks["default_addr"]),
                     }
                     for i in range(nr):
-                        kargs[f"lower_{i}"] = conversion[f"lower_{i}"]
-                        kargs[f"upper_{i}"] = conversion[f"upper_{i}"]
-                        kargs[f"text_{i}"] = conversion.referenced_blocks[f"text_{i}"]
+                        v4_kwargs[f"lower_{i}"] = conversion[f"lower_{i}"]  # type: ignore[literal-required]
+                        v4_kwargs[f"upper_{i}"] = conversion[f"upper_{i}"]  # type: ignore[literal-required]
+                        v4_kwargs[f"text_{i}"] = conversion.referenced_blocks[f"text_{i}"]  # type: ignore[literal-required]
 
-                    conversion = v4b.ChannelConversion(**kargs)
+                    conversion = v4b.ChannelConversion(**v4_kwargs)
 
-            conversion.unit = unit
+            conversion.unit = unit_str
 
     return conversion
 
 
-def inverse_conversion(conversion: ChannelConversionType | dict | None) -> v4b.ChannelConversion | None:
+def inverse_conversion(
+    conversion: (
+        ChannelConversionType | v3b.ChannelConversionKwargs | v4b.ChannelConversionKwargs | dict[str, object] | None
+    ),
+) -> v4b.ChannelConversion | None:
+    if not conversion:
+        return None
 
     if isinstance(conversion, v3b.ChannelConversion):
         conversion = conversion_transfer(conversion, version=4)
 
-    if conversion:
-        if not isinstance(conversion, dict):
-            conversion = to_dict(conversion) or {}
-
-        if "a" in conversion:
-            conv = {
-                "conversion_type": v4c.CONVERSION_TYPE_LIN,
-                "a": 1 / conversion["a"],
-                "b": conversion["b"] / conversion["a"],
-            }
-            conv = v4b.ChannelConversion(**conv)
-
-        elif "P1" in conversion:
-            a, b, c, d, e, f = (conversion[f"P{i}"] for i in range(1, 7))
-
-            if e == 0 and f == 0:
-                if d == 0 and a == 0:
-                    conv = None
-                else:
-                    conv = {
-                        "P1": 0,
-                        "P2": e,
-                        "P3": -b,
-                        "P4": 0,
-                        "P5": -d,
-                        "P6": a,
-                        "conversion_type": v4c.CONVERSION_TYPE_RAT,
-                    }
-                    conv = v4b.ChannelConversion(**conv)
-
-            elif a == 0 and d == 0:
+    conversion_dict: v3b.ChannelConversionKwargs | v4b.ChannelConversionKwargs | dict[str, object]
+    if not isinstance(conversion, dict):
+        conversion_dict = to_dict(conversion)
+    else:
+        conversion_dict = conversion
+
+    if "a" in conversion_dict:
```

Comment:
this will raise an exception if `inverse_conversion` is called with `conversion=None`

Reply:
Check is already done line 282 with an early return line 283 in case it is None. I did this change to reduce code nesting

Reply:
I missed that line, sorry for the false alarm